crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -191,12 +175,12 @@ impl<'w> EntityMut<'w> {
     ///   may happen from **any** `insert_component`, `remove_component` or `despawn`
     ///   operation on this world (non-exhaustive list).
     #[inline]
-    pub unsafe fn get_unchecked_mut<T: Component>(&self) -> Option<Mut<'w, T>> {
+    pub unsafe fn get_unchecked_mut<T: Component>(&self) -> Option<Mut<'_, T>> {
         get_component_and_ticks_with_type(self.world, TypeId::of::<T>(), self.entity, self.location)
             .map(|(value, ticks)| Mut {
-                value: &mut *value.cast::<T>(),
+                value: value.assert_unique().deref_mut::<T>(),
```

Comment:
This should have a comment about the assertion, so it doesn't get optimized out.

Reply:
there's nothing to "Optimize" out its not a real assertion ,its the equivelent of doing `&UnsafeCell<T>` -> `&mut T`

Reply:
Ah I see, it's the only route?